front/graph.send.php

```diff
@@ -30,25 +30,25 @@
  * ---------------------------------------------------------------------
  */
 
-include ('../inc/includes.php');
-
-Session ::checkLoginUser();
+use Glpi\Csv\CsvResponse;
+use Glpi\Csv\StatCsvExport;
+use Glpi\Stat\StatData;
 
-if (isset($_GET["switchto"])) {
-   $_SESSION['glpigraphtype'] = $_GET["switchto"];
-   Html::back();
-}
+include ('../inc/includes.php');
 
-if (($uid = Session::getLoginUserID(false))
-    && isset($_GET["file"])) {
+// Check rights
+Session::checkRight("statistic", READ);
```

Comment:
for this change of right check (before it seems we just checked session), i approve but i'm not very sure to forget a place in glpi self-service where this file could be called

Reply:
All modified stats were only accessible through front files like `stat.location.php` which use the same `Session::checkRight("statistic", READ);` right check.